src/GafferScene/IECoreGLPreview/Renderer.cpp

```diff
@@ -718,13 +722,19 @@ class OpenGLRenderer final : public IECoreScenePreview::Renderer
 			);
 		}
 
-		void renderObjects( IECoreGL::State *currentState )
+		void renderObjects( IECoreGL::State *currentState,
+			boost::container::flat_set<OpenGLObject*> suppressions = boost::container::flat_set<OpenGLObject*>() )
 		{
 			IECoreGL::Selector *selector = IECoreGL::Selector::currentSelector();
 
 			GLuint i = 1;
 			for( const auto &o : m_objects )
 			{
+				if( suppressions.find( o.get() ) != suppressions.end() )
```

Comment:
From a performance perspective, I'm a bit wary of having this in the inner loop for interactive rendering. What do you think to just erasing `camera` from `m_objects` in `renderBatch()`?

Reply:
@danieldresser, I think we agreed in a separate conversation that this is the approach that we'd take. Will you get a chance to do it or would you like me to take it?